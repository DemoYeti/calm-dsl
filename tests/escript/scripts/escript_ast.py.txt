# python3;success

import ast
code_string = """
def add(a, b):
    return a + b

result = add(1, 2)
print(result)
"""
tree = ast.parse(code_string)
print(ast.dump(tree, indent=2))