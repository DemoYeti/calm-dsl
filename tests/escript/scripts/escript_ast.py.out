Module(
  body=[
    FunctionDef(
      name='add',
      args=arguments(
        posonlyargs=[],
        args=[
          arg(arg='a'),
          arg(arg='b')],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      body=[
        Return(
          value=BinOp(
            left=Name(id='a', ctx=Load()),
            op=Add(),
            right=Name(id='b', ctx=Load())))],
      decorator_list=[]),
    Assign(
      targets=[
        Name(id='result', ctx=Store())],
      value=Call(
        func=Name(id='add', ctx=Load()),
        args=[
          Constant(value=1),
          Constant(value=2)],
        keywords=[])),
    Expr(
      value=Call(
        func=Name(id='print', ctx=Load()),
        args=[
          Name(id='result', ctx=Load())],
        keywords=[]))],
  type_ignores=[])